var/spack/repos/builtin/packages/foam-extend/assets/spack-Allwmake

```diff
@@ -0,0 +1,22 @@
+#!/bin/bash
+# Build wrapper script - FOAM_INST_DIR is required by foam-extend
+export FOAM_INST_DIR=$(cd .. && pwd -L)
+. $PWD/etc/bashrc ''                            # No arguments
+mkdir -p $FOAM_APPBIN $FOAM_LIBBIN 2>/dev/null  # Allow interrupt
+echo "Build openfoam with SPACK (WM_NCOMPPROCS=$WM_NCOMPPROCS, $@)"
+echo WM_PROJECT_DIR = $WM_PROJECT_DIR
+./Allwmake $@                                   # Pass arguments
+
+# Link non-dummy MPI_FOAM type to parent-dir, where rpath can find it
+if [ "${FOAM_MPI:=dummy}" != dummy -a -d "$FOAM_LIBBIN/$FOAM_MPI" ]
+then
+(
+    cd "$FOAM_LIBBIN" || exit 1
+    for i in $FOAM_MPI/lib*.so
+    do
+        [ -f $i ] && ln -sf $i "${i##*/}"
+    done
+)
+fi
+
+# -----------------------------------------------------------------------------
```

Comment:
How it was done before was by creating a `EnvironmentModifications` en applying it to the current environment, this can be done in the `install` function.

```
def get_openfoam_environment(self):
        return EnvironmentModifications.from_sourcing_files(
            join_path(self.stage.source_path, 'etc/bashrc'))
```
That can be used anywhere like
```
env_openfoam = self.get_openfoam_environment()
env_openfoam.apply_modifications()
```